frontend/src/components/widgets/DataFrame/columns/utils.ts

```diff
@@ -0,0 +1,358 @@
+ * @param faded: If true, returns a faded version of the cell.
+ *
+ * @return a GridCell object that can be used by glide-data-grid.
+ */
+export function getTextCell(readonly: boolean, faded: boolean): TextCell {
+  const style = faded ? "faded" : "normal"
+  return {
+    kind: GridCellKind.Text,
+    data: "",
+    displayData: "",
+    allowOverlay: true,
+    readonly,
+    style,
+  } as TextCell
+}
+
+/**
+ * Converts from our BaseColumn format to the glide-data-grid compatible GridColumn.
+ */
+export function toGlideColumn(column: BaseColumn): GridColumn {
+  return {
+    id: column.id,
+    title: column.title,
+    hasMenu: false,
+    themeOverride: column.themeOverride,
+    ...(column.isStretched && {
+      grow: column.isIndex ? 1 : 3,
+    }),
+    ...(column.width && {
+      width: column.width,
+    }),
+  } as GridColumn
+}
+
+/**
+ * Merges the default column parameters with the user-defined column parameters.
+ *
+ * @param defaultParams - The default column parameters.
+ * @param userParams - The user-defined column parameters.
+ *
+ * @returns The merged column parameters.
+ */
+export function mergeColumnParameters(
+  defaultParams: Record<string, any> | undefined | null,
+  userParams: Record<string, any> | undefined | null
+): Record<string, any> {
+  if (isNullOrUndefined(defaultParams)) {
+    return userParams || {}
+  }
+
+  if (isNullOrUndefined(userParams)) {
+    return defaultParams || {}
+  }
+
+  return merge(defaultParams, userParams)
+}
+
+/**
+ * Converts the given value of unknown type to an array without
+ * the risks of any exceptions.
+ *
+ * @param data - The value to convert to an array.
+ *
+ * @returns The converted array or an empty array if the value cannot be interpreted as an array.
+ */
+export function toSafeArray(data: any): any[] {
+  if (isNullOrUndefined(data)) {
+    return []
+  }
+
+  if (typeof data === "number" || typeof data === "boolean") {
+    // Single number or boolean
+    return [data]
+  }
+
+  if (typeof data === "string") {
+    if (data === "") {
+      // Empty string
+      return []
+    }
+
+    // Try to parse string to an array
+    if (data.trim().startsWith("[") && data.trim().endsWith("]")) {
+      // Support for JSON arrays: ["foo", 1, null, "test"]
+      try {
+        return JSON.parse(data)
+      } catch (error) {
+        return [data]
+      }
+    } else {
+      // Support for comma-separated values: "foo,1,,test"
+      return data.split(",")
+    }
+  }
+
+  try {
+    const parsedData = JSON.parse(
+      JSON.stringify(data, (_key, value) =>
+        typeof value === "bigint" ? Number(value) : value
+      )
+    )
+    if (!isArray(parsedData)) {
+      return [toSafeString(parsedData)]
+    }
+
+    return parsedData.map((value: any) =>
+      ["string", "number", "boolean", "null"].includes(typeof value)
+        ? value
+        : toSafeString(value)
+    )
+  } catch (error) {
+    return [toSafeString(data)]
```

Comment:
Should this include the error? I feel like it would be a nice to have as a developer.

Reply:
Hmm, good question. These `toSafe..` methods are supposed to do their best to show the value - or information about the value - without having to throw or return an error. Therefore, I think this is better(and more end-user friendly) not to add any error information here. The developer itself will see that something is wrong with the data since it is only rendered as a single list item. I expect that there are cases where this leads to problems. Which eventually will lead to some  Github issues allowing us manually investigate the problems.